roles/simulator/tasks/start.yml

```diff
@@ -1,38 +1,43 @@
 ---
 
-- name: copy config files to ueransim host
+- name: check if servers and config server length are equal
+  fail:
+    msg: "Number of ueransim nodes and servers in config file are not equal"
+  when: ueransim.servers|length != groups['ueransim_nodes']|length
+
+- name: copy gNb config files to ueransim host
   template:
-    src: "{{ ROOT_DIR }}/{{ item.1 }}"
-    dest: ueransim_onramp/config/
-  with_indexed_items: "{{ ueransim.servers[lookup('ansible.utils.index_of', groups['ueransim_nodes'], 'eq', inventory_hostname)] }}"
+    src: "{{ ROOT_DIR }}/{{ ueransim.servers[server_index].gnb }}"
+    dest: ueransim_onramp/config/custom-gnb.yaml
+  vars:
+    server_index: "{{ groups['ueransim_nodes'].index(inventory_hostname) }}"  
+  when: inventory_hostname in groups['ueransim_nodes']
+
+- name: copy UE config files to ueransim host
+  template:
+    src: "{{ ROOT_DIR }}/{{ ueransim.servers[server_index].ue }}"
+    dest: ueransim_onramp/config/custom-ue.yaml
+  vars:
+    server_index: "{{ groups['ueransim_nodes'].index(inventory_hostname) }}"  
   when: inventory_hostname in groups['ueransim_nodes']
```

Comment:
FYI, as it was before, there was no need to have 2 tasks (one for the `ue` and one for the `gnb`). Do you have to create 2 tasks? or is it to make easier to visualize?

Reply:
Initially, the path was set to `config/filename`, but in OnRamp, we follow the convention `deps/ueransim/config/filename`. It wasn't clear which file was for the UE and which was for the gNb. To resolve this, I added tags and now copy each file, renaming them accordingly (for UE and gNb) in a separate loop.

Reply:
Ok, thanks!